packages/ember-data/lib/system/model/states.js

```diff
@@ -417,6 +417,10 @@ createdState.uncommitted.rollback = function(record) {
   record.transitionTo('deleted.saved');
 };
 
+createdState.uncommitted.pushedData = function(record) {
+  record.transitionTo('loaded.saved');
+};
+
```

Comment:
Heh, all my tests show that records inside `buildRecord` leave in the `root.empty` state... which means that your hook won't be invoked anyway. On a related note, it doesn't look like `buildRecord` is ever invoked with its third param, so [this line](https://github.com/emberjs/data/blob/v1.0.0-beta.14.1/packages/ember-data/lib/system/store.js#L1602) will never get executed anyway.